sql/src/main/java/io/crate/analyze/relations/RelationAnalyzer.java

```diff
@@ -146,27 +150,34 @@ protected AnalyzedRelation visitQuerySpecification(QuerySpecification node, Rela
                 .groupBy(groupBy)
                 .hasAggregates(expressionAnalysisContext.hasAggregates);
 
+        QueriedRelation relation = null;
         if (context.sources().size() == 1) {
             AnalyzedRelation source = Iterables.getOnlyElement(context.sources().values());
-            QueriedTableRelation relation;
+            QueriedTableRelation tableRelation = null;
             if (source instanceof DocTableRelation) {
-                relation = new QueriedDocTable(
-                        (DocTableRelation) source, selectAnalysis.outputNames(), querySpec);
+                tableRelation = new QueriedDocTable((DocTableRelation) source, selectAnalysis.outputNames(), querySpec);
             } else if (source instanceof TableRelation) {
-                relation =  new QueriedTable((TableRelation) source, selectAnalysis.outputNames(), querySpec);
+                tableRelation = new QueriedTable((TableRelation) source, selectAnalysis.outputNames(), querySpec);
             } else {
-                throw new UnsupportedOperationException("Only tables are allowed in the FROM clause, got: " + source);
+                assert source instanceof QueriedRelation : "expecting relation to be an instance of QueriedRelation";
+                relation = new QueriedSelectRelation((QueriedRelation) source, selectAnalysis.outputNames(), querySpec);
             }
-            relation.normalize(analysisMetaData);
-            return relation;
+            if (tableRelation != null) {
+                tableRelation.normalize(analysisMetaData);
+                relation = tableRelation;
+            }
+        } else {
+            // TODO: implement multi table selects
```

Comment:
Good question, this is really old. I think that what was meant by this comment is that we need to call `MultiSourceSelect#normalize` for the querySpec. @dobe is that it? I don't know the implications of not doing so since all tests are passing without it.